sesame/src/main/scala/org/w3/banana/sesame/SesameStore.scala

```diff
@@ -15,9 +15,23 @@ class SesameStore
 
   /* Transactor */
 
-  def r[T](conn: RepositoryConnection, body: => T): Try[T] = ???
+  def r[T](conn: RepositoryConnection, body: => T): Try[T] = Try{
+    val result = body
+    conn.close()
+    result
+  }
 
-  def rw[T](conn: RepositoryConnection, body: => T): Try[T] = ???
+  def rw[T](conn: RepositoryConnection, body: => T): Try[T] = Try{
+    conn.begin()
+    val result = body
+    conn.commit()
+    conn.close()
+    result
+  }.recoverWith{ case exception=>
```

Comment:
After some consideration, I came to a conclusion that try / catch does not make sense here (to be honest I never use them in my code, as I consider it a java-way of handling errors). I have to anyway wrap everything in Try and I also have to make a rollback after any error, so recoverWith looks suitable here